### ISSUE TITLE
Avoid recomposition on object movement

### ISSUE THREAD

**Issue by mtpdog** [ramani-maps/ramani-maps#60]
I've done a stress test adding 10,000 circles. Ramani Maps performs really well while moving and zooming the map.
However, I've noticed that the full recomposition is fired when any element is dragged. 
Is it possible to not call the full recomposition and redraw only that element?

https://github.com/ramani-maps/ramani-maps/assets/150581563/97344fa6-cb91-4d89-beb6-7e50d503c762

https://github.com/ramani-maps/ramani-maps/assets/150581563/a82f7200-3f4f-4491-abd7-69b93306f15e

Test code to add 10,000 circles:
`val TOTAL_CIRCLES = 10000
                    for (i in 0..TOTAL_CIRCLES){
                        Circle(
                            center = LatLng(44.986 + Math.random()/10, 10.812 + Math.random()/10),
                            radius = 2.0F,
                            color = "Green",
                            zIndex = 1,
                            isDraggable = true,
                        )
                    }`

**Comment by mtpdog** [comment]
Btw, recomposition is also fired when "Center on polygon" button is clicked (= camera "Fly" movement is performed).

**Comment by JonasVautherin** [comment]
Good point, it would be nice to investigate! Let us know if you find something, otherwise I'll have a look when I have time :+1:.

**Comment by JonasVautherin** [comment]
I finally looked into it, and I believe that this is not a problem in Ramani but in the calling code.

For instance, clicking "Center on polygon" will recompose everything that is below the state of [`cameraPosition`](https://github.com/ramani-maps/ramani-maps/blob/main/examples/interactive-polygon/app/src/main/java/org/ramani/example/interactive_polygon/MainActivity.kt#L33-L35), right?

I would say that the code has to be written in such a way that the state modified when dragging the polygon does not trigger a recomposition of the 10k circles.

I did that in an example and it works well.

Or do you think I am missing something?

**Comment by JonasVautherin** [comment]
Feel free to reopen if necessary, but it feels like this can be closed now :+1:.
